### PR TITLE
feat: add manual input definitions for sector II

### PR DESCRIPTION
**Pull request by evanp**
Please review and merge!

### CODE REVIEW

File: app/src/util/form-schema/manual-input-hierarchy.json [hunk 1]
```diff
@@ -1467,10 +1522,25 @@
...
+            "on-road-transport-vehicle-type": "vehicle-type-public-transport-vehicles"
+          }
         ]
       }
     ],
     "directMeasure": {
+      "id": "on-road-transport-direct-measure-energy-consumption-methodology",
       "extra-fields": [
         {
           "id": "on-road-transport-vehicle-type",
```
Comment (mfonsecaOEF) [inline app/src/util/form-schema/manual-input-hierarchy.json:1546]: in "on-road-transport-fuel-type" is missing:
- the option: "fuel-type-all"
- "exclusive": "fuel-type-all"

File: app/src/util/form-schema/manual-input-hierarchy.json [hunk 2]
```diff
@@ -1389,26 +1355,114 @@
...
+                  "fuel-type-lpg",
+                  "fuel-type-cng",
+                  "fuel-type-biofuel"
+                ]
+              },
+              {
+                "id": "on-road-transport-fuel-intensity",
+                "type": "number",
+                "units": [
+                  "units-grams-per-kilometer",
```
Comment (mfonsecaOEF) [inline app/src/util/form-schema/manual-input-hierarchy.json:1433]: units for fuel intensity:
- tonnes-per-kilometer
- gallons-per-kilometer
- cubic-meters-per-kilometer
- tonnes-per-miles
- gallons-per-miles
- cubic-meters-per-miles

**Review by lemilonkh — APPROVED** [review]

File: app/src/util/form-schema/manual-input-hierarchy.json [hunk 3]
```diff
@@ -1490,6 +1560,835 @@
...
+              },
+              {
+                "id": "railways-vehicle-type",
+                "options": [
+                  "vehicle-type-all",
+                  "vehicle-type-passenger-trains",
+                  "vehicle-type-freight-trains",
+                  "vehicle-type-high-speed-trains",
+                  "vehicle-type-tourist-trains"
+                ]
```
Comment (mfonsecaOEF) [inline app/src/util/form-schema/manual-input-hierarchy.json:1598]: "exclusive": "vehicle-type-all"

File: app/src/util/form-schema/manual-input-hierarchy.json [hunk 4]
```diff
@@ -1490,6 +1560,835 @@
...
+            "units": ["units-kilometers", "units-miles"],
+            "extra-fields": [
+              {
+                "id": "railways-vehicle-type",
+                "options": [
+                  "vehicle-type-all",
+                  "vehicle-type-passenger-trains",
+                  "vehicle-type-freight-trains",
+                  "vehicle-type-high-speed-trains",
+                  "vehicle-type-tourist-trains"
```
Comment (mfonsecaOEF) [inline app/src/util/form-schema/manual-input-hierarchy.json:1650]: "exclusive": "vehicle-type-all"

File: app/src/util/form-schema/manual-input-hierarchy.json [hunk 5]
```diff
@@ -1490,6 +1560,835 @@
...
+                  "fuel-type-lpg",
+                  "fuel-type-cng",
+                  "fuel-type-biofuel"
+                ]
+              },
+              {
+                "id": "railways-fuel-intensity",
+                "type": "number",
+                "units": [
+                  "units-grams-per-kilometer",
```
Comment (mfonsecaOEF) [inline app/src/util/form-schema/manual-input-hierarchy.json:1667]: same here with the units

File: app/src/util/form-schema/manual-input-hierarchy.json [hunk 6]
```diff
@@ -1490,6 +1560,835 @@
...
+        {
+          "id": "railways-vehicle-type",
+          "options": [
+            "vehicle-type-all",
+            "vehicle-type-passenger-trains",
+            "vehicle-type-freight-trains",
+            "vehicle-type-high-speed-trains",
+            "vehicle-type-tourist-trains"
+          ],
+          "multiselect": true
```
Comment (mfonsecaOEF) [inline app/src/util/form-schema/manual-input-hierarchy.json:1716]: "exclusive": "vehicle-type-all"

File: app/src/util/form-schema/manual-input-hierarchy.json [hunk 7]
```diff
@@ -1490,6 +1560,835 @@
...
+                "type": "text"
+              }
+            ],
+            "units": [
+              "units-gallons",
+              "units-liters",
+              "units-cubic-meters",
+              "units-kilograms",
+              "units-terajoules",
+              "units-kilowatt-hours"
```
Comment (mfonsecaOEF) [inline app/src/util/form-schema/manual-input-hierarchy.json:1611]: drop:
- "units-kilowatt-hours"

File: app/src/util/form-schema/manual-input-hierarchy.json [hunk 8]
```diff
@@ -1490,6 +1560,835 @@
...
+            "id": "electricity-consumption-railways-activity",
+            "unique-by": [
+              "railways-vehicle-type",
+              "railways-electricity-source"
+            ],
+            "extra-fields": [
+              {
+                "id": "railways-vehicle-type",
+                "options": [
+                  "vehicle-type-all",
```
Comment (mfonsecaOEF) [inline app/src/util/form-schema/manual-input-hierarchy.json:1740]: "exclusive": "vehicle-type-all"

File: app/src/util/form-schema/manual-input-hierarchy.json [hunk 9]
```diff
@@ -1490,6 +1560,835 @@
...
+          }
+        ]
+      }
+    ],
+    "directMeasure": {
+      "id": "railways-direct-measure-energy-consumption-methodology",
+      "extra-fields": [
+        {
+          "id": "railways-vehicle-type",
+          "options": [
```
Comment (mfonsecaOEF) [inline app/src/util/form-schema/manual-input-hierarchy.json:1775]: missing options:
the option: "vehicle-type-all"
"exclusive": "vehicle-type-all"

File: app/src/util/form-schema/manual-input-hierarchy.json [hunk 10]
```diff
@@ -1490,6 +1560,835 @@
...
+              },
+              {
+                "id": "waterborne-navigation-vehicle-type",
+                "options": [
+                  "vehicle-type-all",
+                  "vehicle-type-ferries",
+                  "vehicle-type-boats",
+                  "vehicle-type-marine-vessels"
+                ]
+              },
```
Comment (mfonsecaOEF) [inline app/src/util/form-schema/manual-input-hierarchy.json:1824]: "exclusive": "vehicle-type-all"

File: app/src/util/form-schema/manual-input-hierarchy.json [hunk 11]
```diff
@@ -1490,6 +1560,835 @@
...
+            "extra-fields": [
+              {
+                "id": "waterborne-navigation-vehicle-type",
+                "options": [
+                  "vehicle-type-all",
+                  "vehicle-type-ferries",
+                  "vehicle-type-boats",
+                  "vehicle-type-marine-vessels"
+                ]
+              },
```
Comment (mfonsecaOEF) [inline app/src/util/form-schema/manual-input-hierarchy.json:1879]: "exclusive": "vehicle-type-all"

File: app/src/util/form-schema/manual-input-hierarchy.json [hunk 12]
```diff
@@ -1490,6 +1560,835 @@
...
+            "fuel-type-diesel",
+            "fuel-type-lpg",
+            "fuel-type-cng",
+            "fuel-type-biofuel"
+          ]
+        },
+        {
+          "id": "waterborne-navigation-vehicle-type",
+          "options": [
+            "vehicle-type-ferries",
```
Comment (mfonsecaOEF) [inline app/src/util/form-schema/manual-input-hierarchy.json:1933]: missing options:
the option: "vehicle-type-all"
"exclusive": "vehicle-type-all"

File: app/src/util/form-schema/manual-input-hierarchy.json [hunk 13]
```diff
@@ -1490,6 +1560,835 @@
...
+              "waterborne-navigation-vehicle-type",
+              "waterborne-navigation-electricity-source"
+            ],
+            "extra-fields": [
+              {
+                "id": "waterborne-navigation-vehicle-type",
+                "options": [
+                  "vehicle-type-all",
+                  "vehicle-type-ferries",
+                  "vehicle-type-boats",
```
Comment (mfonsecaOEF) [inline app/src/util/form-schema/manual-input-hierarchy.json:1963]: "exclusive": "vehicle-type-all"

File: app/src/util/form-schema/manual-input-hierarchy.json [hunk 14]
```diff
@@ -1490,6 +1560,835 @@
...
+        ]
+      }
+    ],
+    "directMeasure": {
+      "id": "waterborne-navigation-direct-measure-energy-consumption-methodology",
+      "extra-fields": [
+        {
+          "id": "waterborne-navigation-vehicle-type",
+          "options": [
+            "vehicle-type-ferries",
```
Comment (mfonsecaOEF) [inline app/src/util/form-schema/manual-input-hierarchy.json:1996]: missing options:
the option: "vehicle-type-all"
"exclusive": "vehicle-type-all"

File: app/src/util/form-schema/manual-input-hierarchy.json [hunk 15]
```diff
@@ -1490,6 +1560,835 @@
...
+              "aviation-fuel-type",
+              "aviation-fuel-source"
+            ],
+            "activity-title": "activity-total-fuel-sales",
+            "minimum": 1,
+            "extra-fields": [
+              {
+                "id": "aviation-fuel-type",
+                "options": [
+                  "fuel-type-gasoline",
```
Comment (mfonsecaOEF) [inline app/src/util/form-schema/manual-input-hierarchy.json:2028]: the correct fuel list:
- Aviation Gasoline
- Diesel
- Jet Gasoline
- Jet Kerosene
- Liquefied Petroleum Gas
- Gasoline
- Natural Gas Liquids

the change was applied in figma as well.

File: app/src/util/form-schema/manual-input-hierarchy.json [hunk 16]
```diff
@@ -1490,6 +1560,835 @@
...
+                ]
+              },
+              {
+                "id": "aviation-vehicle-type",
+                "options": [
+                  "vehicle-type-all",
+                  "vehicle-type-passenger-aircraft",
+                  "vehicle-type-cargo-aircraft",
+                  "vehicle-type-private-and-business-aircraft",
+                  "vehicle-type-helicopters"
```
Comment (mfonsecaOEF) [inline app/src/util/form-schema/manual-input-hierarchy.json:2042]: "exclusive": "vehicle-type-all"

File: app/src/util/form-schema/manual-input-hierarchy.json [hunk 17]
```diff
@@ -1490,6 +1560,835 @@
...
+            "extra-fields": [
+              {
+                "id": "aviation-vehicle-type",
+                "options": [
+                  "vehicle-type-all",
+                  "vehicle-type-passenger-aircraft",
+                  "vehicle-type-cargo-aircraft",
+                  "vehicle-type-private-and-business-aircraft",
+                  "vehicle-type-helicopters"
+                ]
```
Comment (mfonsecaOEF) [inline app/src/util/form-schema/manual-input-hierarchy.json:2096]: "exclusive": "vehicle-type-all"

File: app/src/util/form-schema/manual-input-hierarchy.json [hunk 18]
```diff
@@ -1490,6 +1560,835 @@
...
+            "fuel-type-diesel",
+            "fuel-type-lpg",
+            "fuel-type-cng",
+            "fuel-type-biofuel"
+          ]
+        },
+        {
+          "id": "aviation-vehicle-type",
+          "options": [
+            "vehicle-type-passenger-aircraft",
```
Comment (mfonsecaOEF) [inline app/src/util/form-schema/manual-input-hierarchy.json:2151]: missing options:
the option: "vehicle-type-all"
"exclusive": "vehicle-type-all"

File: app/src/util/form-schema/manual-input-hierarchy.json [hunk 19]
```diff
@@ -1490,6 +1560,835 @@
...
+      "id": "aviation-direct-measure-methodology",
+      "extra-fields": [
+        {
+          "id": "aviation-fuel-type",
+          "options": [
+            "fuel-type-gasoline",
+            "fuel-type-diesel",
+            "fuel-type-lpg",
+            "fuel-type-cng",
+            "fuel-type-biofuel"
```
Comment (mfonsecaOEF) [inline app/src/util/form-schema/manual-input-hierarchy.json:2145]: missing options:
the option: "fuel-type-all"
"exclusive": "fuel-type-all"

File: app/src/util/form-schema/manual-input-hierarchy.json [hunk 20]
```diff
@@ -1490,6 +1560,835 @@
...
+                ]
+              },
+              {
+                "id": "off-road-transport-vehicle-type",
+                "options": [
+                  "vehicle-type-all",
+                  "vehicle-type-construction-machinery",
+                  "vehicle-type-agricultural-machinery",
+                  "vehicle-type-mining-equipment",
+                  "vehicle-type-forestry-equipment"
```
Comment (mfonsecaOEF) [inline app/src/util/form-schema/manual-input-hierarchy.json:2264]: "exclusive": "vehicle-type-all"

File: app/src/util/form-schema/manual-input-hierarchy.json [hunk 21]
```diff
@@ -1490,6 +1560,835 @@
...
+            "off-road-transport-vehicle-type": "vehicle-type-agricultural-machinery"
+          }
+        ]
+      }
+    ],
+    "directMeasure": {
+      "extra-fields": [
+        {
+          "id": "off-road-transport-fuel-type",
+          "options": [
```
Comment (mfonsecaOEF) [inline app/src/util/form-schema/manual-input-hierarchy.json:2302]: missing option:
fuel-type-all
"exclusive": "fuel-type-all"

File: app/src/util/form-schema/manual-input-hierarchy.json [hunk 22]
```diff
@@ -1490,6 +1560,835 @@
...
+            "fuel-type-diesel",
+            "fuel-type-lpg",
+            "fuel-type-cng",
+            "fuel-type-biofuel"
+          ]
+        },
+        {
+          "id": "off-road-transport-vehicle-type",
+          "options": [
+            "vehicle-type-construction-machinery",
```
Comment (mfonsecaOEF) [inline app/src/util/form-schema/manual-input-hierarchy.json:2313]: missing option:
vehicle-type-all
"exclusive": "vehicle-type-all"

File: app/src/util/form-schema/manual-input-hierarchy.json [hunk 23]
```diff
@@ -1490,6 +1560,835 @@
...
+            ],
+            "extra-fields": [
+              {
+                "id": "off-road-transport-vehicle-type",
+                "options": [
+                  "vehicle-type-all",
+                  "vehicle-type-construction-machinery",
+                  "vehicle-type-agricultural-machinery",
+                  "vehicle-type-mining-equipment",
+                  "vehicle-type-forestry-equipment"
```
Comment (mfonsecaOEF) [inline app/src/util/form-schema/manual-input-hierarchy.json:2347]: "exclusive": "vehicle-type-all"

File: app/src/util/form-schema/manual-input-hierarchy.json [hunk 24]
```diff
@@ -1490,6 +1560,835 @@
...
+          }
+        ]
+      }
+    ],
+    "directMeasure": {
+      "extra-fields": [
+        {
+          "id": "off-road-transport-vehicle-type",
+          "options": [
+            "vehicle-type-construction-machinery",
```
Comment (mfonsecaOEF) [inline app/src/util/form-schema/manual-input-hierarchy.json:2378]: missing option:
vehicle-type-all
"exclusive": "vehicle-type-all"

File: app/src/util/form-schema/manual-input-hierarchy.json [hunk 25]
```diff
@@ -1389,26 +1355,114 @@
...
           }
+        ],
+        "suggestedActivities": [
+          {
+            "id": "fuel-sales-on-road-transport-activity",
+            "on-road-transport-vehicle-type": "vehicle-type-all"
+          },
+          {
+            "id": "fuel-sales-on-road-transport-activity",
+            "on-road-transport-vehicle-type": "vehicle-type-passenger-vehicles"
```
[the remark this answers is not in the document]
Comment (evanp) [reply app/src/util/form-schema/manual-input-hierarchy.json:1366]: > drop repeated id rows and add a "suggestedActivitiesId", e.g: "suggestedActivitiesId": "fuel-sales-on-road-transport-activity-type", "suggestedActivities": [ { "id": "vehicle-type-all" }, { "id": "vehicle-type-passenger-vehicle" }, { "id": "vehicle-type-public-transport-vehicles" } ]
> 
> suggestion from Nina

This format is so we can have pre-filled forms to pop up.

The "id" is for the activity form to pop up.

The other properties are for the pre-filled values.

I don't think we've confirmed that each methodology will only have one kind of activity form to show, so for now, I'd like to keep this structure as it stands.

**Pull request merged**